src/Parser/RawBody.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace React\Http\Parser;
+
+use Evenement\EventEmitterTrait;
+use React\Http\Request;
+
+class RawBody implements ParserInterface
+{
+    use EventEmitterTrait;
+
+    protected $buffer = '';
+    protected $contentLength;
+
+    public function __construct(Request $request)
+    {
+        $headers = $request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        $this->contentLength = $headers['content-length'];
```

Comment:
Note that RFC 2616 is deprecated, see http://tools.ietf.org/html/rfc7230 and http://tools.ietf.org/html/rfc7231 instead.

> My application should only get valid requests.

I agree, this should be our main focus here :+1: 

Invalid requests should be rejected as per the spec, handling this as a consumer of this lib is probably out of scope.

My main issue with this ticket is that it tries to tackle multiple responsibilities. HTTP is a layered protocol and the newer specs make this much more obvious: Message framing (content-length, transfer-encoding etc.) should be handled *before* passing anything to the form parsers.